build/release.sh

```diff
@@ -15,18 +15,56 @@ if [ -z $1 ] || [ -z $2 ] || [[ $2 != "build" && $2 != "publish" && $2 != "build
 fi
 
 VERSION=$1
-TAG=geoserver-docker.osgeo.org/geoserver:$VERSION
+MAIN="2.24"
+if [[ "${VERSION:0:4}" == "$MAIN" ]]; then
+  # main branch snapshot release
+  BRANCH=main
+  TAG=geoserver-docker.osgeo.org/geoserver:$MAIN.x
+else
+  if [[ "$VERSION" == *"-SNAPSHOT"* ]]; then
+    # stable or maintenance branch snapshot release
+    BRANCH="${VERSION:0:4}.x"
+    TAG=geoserver-docker.osgeo.org/geoserver:$BRANCH
+  else
+    BRANCH="${VERSION:0:4}.x"
+    TAG=geoserver-docker.osgeo.org/geoserver:$VERSION
+  fi
+fi
+
+echo "Release from branch $BRANCH GeoServer $VERSION as $TAG" 
 
 # Go up one level to the Dockerfile
 cd ".."
 
-if [ $2 != "publish" ]; then
+if [ $2 == *"build"* ]; then
   echo "Building GeoServer Docker Image..."
-  echo "docker build --build-arg GS_VERSION=$VERSION -t $TAG ."
-  docker build --build-arg GS_VERSION=$VERSION -t $TAG .
+  if [[ "$VERSION" == *"-SNAPSHOT"* ]]; then
+    echo "  nightly build from https://build.geoserver.org/geoserver/$BRANCH"
+    echo
+    if [[ "$BRANCH" == "main" ]]; then
+      echo "docker build --build-arg GS_VERSION=$VERSION -t $TAG ."
+      docker build \
+        --build-arg WAR_ZIP_URL=https://build.geoserver.org/geoserver/main/geoserver-main-latest-war.zip \
+        --build-arg STABLE_PLUGIN_URL=https://build.geoserver.org/geoserver/main/ext-latest/ \
+        --build-arg COMMUNITY_PLUGIN_URL=https://build.geoserver.org/geoserver/main/community-latest/ \
+        --build-arg GS_VERSION=$VERSION \
+        -t $TAG .
+    else
+      echo "docker build --build-arg GS_VERSION=$VERSION -t $TAG ."
+      docker build \
+        --build-arg WAR_ZIP_URL=https://build.geoserver.org/geoserver/$BRANCH/geoserver-$BRANCH-latest-war.zip \
+        --build-arg STABLE_PLUGIN_URL=https://build.geoserver.org/geoserver/$BRANCH/ext-latest/ \
+        --build-arg COMMUNITY_PLUGIN_URL=https://build.geoserver.org/geoserver/$BRANCH/community-latest/ \
+        --build-arg GS_VERSION=$VERSION \
+        -t $TAG .
```

Comment:
For ``SNAPSHOT`` release ``WAR_ZIP_URL``,``STABLE_PLUGIN_URL``, and ``COMMUNITY_PLUGIN_URL`` are required to pointing to build.geoserver.org